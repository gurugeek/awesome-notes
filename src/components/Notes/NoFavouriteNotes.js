import React from 'react';
import { View, Text } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import { NotesStyle } from '../../styles/index';

const NoFavouriteNotes = () => {
  return (
    <View style={NotesStyle.emptyNotes.wrapper}>
        <View style={NotesStyle.emptyNotes.iconRow}>
            <Icon name="ios-star-outline" size={50} color="#FDE74C" />
        </View>
        <View style={NotesStyle.emptyNotes.titleRow}>
            <Text style={NotesStyle.emptyNotes.titleText}>You have no favourite notes..yet...</Text>
        </View>
        <View style={NotesStyle.emptyNotes.messageRow}>
            <Text style={NotesStyle.emptyNotes.messageText}>Add your first one swiping your note from left to right and tapping on the <Icon name="ios-star-outline" size={16} color="#FDE74C" /></Text>
        </View>
    </View>
  );
};

export default NoFavouriteNotes;