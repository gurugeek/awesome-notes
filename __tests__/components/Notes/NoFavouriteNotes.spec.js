import React from 'react';
import { View, Text, Animated } from 'react-native';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Icon from 'react-native-vector-icons/Ionicons';

Enzyme.configure({ adapter: new Adapter() });

import NoFavouriteNotes from '../../../src/components/Notes/NoFavouriteNotes';
import { StackNavigator } from 'react-navigation';

jest.useFakeTimers();

describe('<NoFavouriteNotes /> Component', () => {
    it('NoFavouriteNotes should match with snapshot', () => {
        const tree = renderer.create(
            <NoFavouriteNotes />
        );
        const json = tree.toJSON();
        expect(json).toMatchSnapshot();
    });

    const noFavouriteNotesShallow = shallow(<NoFavouriteNotes />);

    it('NoFavouriteNotes should exists', () => {
        expect(noFavouriteNotesShallow.length).toEqual(1);
    });

    it('NoFavouriteNotes should has three View', () => {
        expect(noFavouriteNotesShallow.find(View)).toHaveLength(3);
    });

    it('NoFavouriteNotes should has one Animated.View', () => {
        expect(noFavouriteNotesShallow.find(Animated.View)).toHaveLength(1);
    });

    it('NoFavouriteNotes should has two Text', () => {
        expect(noFavouriteNotesShallow.find(Text)).toHaveLength(2);
    });

    it('NoFavouriteNotes should has two Icon', () => {
        expect(noFavouriteNotesShallow.find(Icon)).toHaveLength(2);
    });
});