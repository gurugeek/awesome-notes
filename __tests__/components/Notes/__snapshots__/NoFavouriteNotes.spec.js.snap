// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoFavouriteNotes /> Component NoFavouriteNotes should match with snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#457B9D",
      "borderRadius": 5,
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FDE74C",
            "fontSize": 50,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 40,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#F1FAEE",
          "fontFamily": "Roboto-Bold",
          "fontSize": 24,
          "textAlign": "center",
        }
      }
    >
      You have no favourite notes..yet...
    </Text>
  </View>
  <View
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#F1FAEE",
          "fontFamily": "Roboto-Regular",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Add your first one swiping your note from left to right and tapping on the 
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#FDE74C",
              "fontSize": 16,
            },
            undefined,
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </Text>
  </View>
</View>
`;
